src/fbjava-impl/src/main/java/org/firebirdsql/fbjava/impl/ExternalProcedure.java

```diff
@@ -86,22 +87,30 @@ public IExternalResultSet open(IStatus status, IExternalContext context, Pointer
 				};
 
 				final ThrowableFunction<Object, IExternalResultSet> postExecute = out -> {
-					final ExternalResultSet rs = (ExternalResultSet) out;
-
-					if (rs == null)
+					if (out == null)
 					{
 						for (int i = inCount; i < inOut.length; ++i)
 							inOut2[i] = Array.get(inOut[i], 0);
 
 						routine.putInMessage(status, context, routine.outputParameters, inOut2, inCount, outMsg);
 						return null;
 					}
-					else
+					else if(ExternalResultSet.class.isAssignableFrom(out.getClass()))
```

Comment:
1) `if (`
2) Why not `instanceof` instead of `isAssignableFrom`?

Reply:
`instanceof` does not work at runtime, only at compile time.

Reply:
You are wrong.

Reply:
@vasiliy-yashkov In the code in question, `out instanceof ExternalResultSet` will do exactly the same. It is not a compile time check, it is a runtime check. You would only need to use `isAssignableFrom` if either `ExternalResultSet.class` would be replaced with something that dynamically determines the class, or instead of `out` you only have a `Class`.

Reply:
> @vasiliy-yashkov In the code in question, `out instanceof ExternalResultSet` will do exactly the same. It is not a compile time check, it is a runtime check. You would only need to use `isAssignableFrom` if either `ExternalResultSet.class` would be replaced with something that dynamically determines the class, or instead of `out` you only have a `Class`.

Thanks for your explanation! I will keep that in mind for the future :)